apps/web/src/components/auth/CreateOrganizationForm.tsx

```diff
@@ -41,6 +46,11 @@ export function CreateOrganization({}: Props) {
       const userData = decode<IJwtPayload>(token);
 
       if (userData.environmentId) {
+        if (isFromVercel) {
```

Comment:
had to manually add checks throughout the auth flow, to start vercel integration and redirect back to the vercel page after successful setup.

I feel, having a proper separate UI elements and pages dedicated to vercel onboarding would help to remove these checks